packages/core/src/internal/services/i18n.service.ts

```diff
@@ -183,17 +184,42 @@ export class I18nService {
   static get keys(): Readonly<I18nStrings> {
     // intialize registry if not yet set
     if (Object.keys(GlobalStateService.state.i18nRegistry).length === 0) {
-      GlobalStateService.state.i18nRegistry = { ...componentStringsDefault, ...GlobalStateService.state.i18nRegistry };
+      GlobalStateService.state.i18nRegistry = mergeObjects(
+        componentStringsDefault,
+        GlobalStateService.state.i18nRegistry
+      );
+    }
+
+    return mergeObjects(componentStringsDefault, GlobalStateService.state.i18nRegistry) as I18nStrings;
+  }
+
+  static findKey(keyValueObj: any) {
+    const keyVals = I18nService.keys;
+    return Object.keys(keyVals).find(key => objectNaiveDeepEquals((keyVals as any)[key], keyValueObj));
+  }
+
+  static get(key: string) {
+    if (!key) {
+      return {};
     }
 
-    return { ...componentStringsDefault, ...GlobalStateService.state.i18nRegistry };
+    return (I18nService.keys as any)[key] || {};
+  }
+
+  static reset() {
```

Comment:
This is used primarily in the unit tests. Other folks could use it if they wanted. 🤷‍♂️

Reply:
We may wan't to mark this one with a `@private`. I'd be worried since this is global state. In microfrontend situation, apps could unintentionally reset the i18n values and wipe out values set by other parts of the application they don't control.